### PR TITLE
chore: add README, nox support for MQTT example

### LINKED ISSUE #188: The demo `cloudiot_mqtt_example.py` doesn't have associated README file with instructions.
#### Overview

I am attempting to run the MQTT demo [here](https://github.com/googleapis/python-iot/blob/main/samples/api-client/mqtt_example/cloudiot_mqtt_example.py) to send some test data to my Cloud IoT Core project. The documentation for the demo script says the following:
>Before you run the sample, you must follow the instructions in the README for this sample.

I would expect the README to contain instructions on the CLI command to run, detailing the arguments to pass in (similar to the [IoT Manager sample](https://github.com/googleapis/python-iot/tree/main/samples/api-client/manager)). However, I could not find any such README. I have had to read the code of the `mqtt_device_demo(args)` function to determine what arguments I needed to pass in.

#### Suggested action

Similar to the [IoT Manager sample](https://github.com/googleapis/python-iot/tree/main/samples/api-client/manager), create the README in the [MQTT sample directory](https://github.com/googleapis/python-iot/tree/main/samples/api-client/mqtt_example) with the appropriate docs for the CLI functions along with example usages.

### PR DESCRIPTION
**Pull request by gcseh**
Fix for #188

### CODE REVIEW

File: samples/api-client/mqtt_example/README.rst [hunk 1]
```diff
@@ -0,0 +1,151 @@
+.. This file is automatically generated. Do not edit this file directly.
+
+Google Cloud IoT Core Python Samples
+===============================================================================
+
+.. image:: https://gstatic.com/cloudssh/images/open-btn.png
+   :target: https://console.cloud.google.com/cloudshell/open?git_repo=https://github.com/GoogleCloudPlatform/python-docs-samples&page=editor&open_in_editor=iot/api-client/mqtt_example/README.rst
```
Comment (dandhlee) [inline samples/api-client/mqtt_example/README.rst:7]: It's pointing to `GoogleCloudPlatform/python-docs-samples` repo but the sample lives right here. I'm hoping this doesn't hardcode to `python-docs-samples` repo.

File: samples/api-client/mqtt_example/README.rst [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+
+This directory contains samples for Google Cloud IoT Core. `Google Cloud IoT Core`_ allows developers to easily integrate Publish and Subscribe functionality with devices and programmatically manage device authorization.
```
Comment (dandhlee) [inline samples/api-client/mqtt_example/README.rst:10]: nit: the first mention to `Google Cloud IoT Core` isn't linked, then the two subsequent ones are linked. Consider either only have the first one linked, or have all of them linked.

File: samples/api-client/mqtt_example/README.rst [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+This example connects a device to `Google Cloud IoT Core`_ via MQTT, using a JWT for authentication. After connecting, the device publishes messages to the device's  MQTT topic at a rate of one per second.
+Note that before you can run this sample, you  must register a device as described  in the parent README. For the gateway samples, you must register and bind a device  as described in the  `Cloud IoT gateway docs <https://cloud.google.com/iot/docs/how-tos/gateways/#setup>`_.
```
Comment (dandhlee) [inline samples/api-client/mqtt_example/README.rst:12]: > in the parent README.

Please put a link to this README.

Comment (dandhlee) [inline samples/api-client/mqtt_example/README.rst:12]: nit: `https://cloud.google.com/iot/docs/how-tos/gateways/#setup` should be `https://cloud.google.com/iot/docs/how-tos/gateways#setup` for correct anchor usage, but please correct me if this is also a style that's used. Does not impact the workflow as it still points to the correct section for Setup, just looked odd to me but I'm happy to learn other styles!

File: samples/api-client/mqtt_example/README.rst [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    https://cloud.google.com/docs/authentication/getting-started
+
+Install Dependencies
+++++++++++++++++++++
+
+#. Clone python-docs-samples and change directory to the sample directory you want to use.
+
+    .. code-block:: bash
+
+        $ git clone https://github.com/GoogleCloudPlatform/python-docs-samples.git
```
Comment (dandhlee) [inline samples/api-client/mqtt_example/README.rst:44]: Please change `python-docs-samples` to `python-iot`.

File: samples/api-client/mqtt_example/README.rst [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+.. _virtualenv: https://virtualenv.pypa.io/
+
+Samples
+-------------------------------------------------------------------------------
+
+MQTT example
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+.. image:: https://gstatic.com/cloudssh/images/open-btn.png
+   :target: https://console.cloud.google.com/cloudshell/open?git_repo=https://github.com/GoogleCloudPlatform/python-docs-samples&page=editor&open_in_editor=iot/api-client/mqtt_example/cloudiot_mqtt_example.py,iot/api-client/mqtt_example/README.rst
```
Comment (dandhlee) [inline samples/api-client/mqtt_example/README.rst:74]: ditto for link to `python-docs-samples`.

File: samples/api-client/mqtt_example/README.rst [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+
+
+
+
+To run this sample:
+
+.. code-block:: bash
```
Comment (dandhlee) [inline samples/api-client/mqtt_example/README.rst:81]: nit: some words are color coded when they shouldn't be, since this is marked in bash. Looks very odd to see random syntax highlighting from place to place, consider changing bash to no-highlight or something more appropriate.

File: samples/api-client/mqtt_example/README.rst [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+                                    [--listen_dur LISTEN_DUR]
+                                    [--message_type {event,state}]
+                                    [--mqtt_bridge_hostname MQTT_BRIDGE_HOSTNAME]
+                                    [--mqtt_bridge_port {8883,443}]
+                                    [--num_messages NUM_MESSAGES]
+                                    --private_key_file PRIVATE_KEY_FILE
+                                    [--project_id PROJECT_ID] --registry_id
+                                    REGISTRY_ID
+                                    [--service_account_json SERVICE_ACCOUNT_JSON]
+                                    {device_demo,gateway_send,gateway_listen} ...
```
Comment (dandhlee) [inline samples/api-client/mqtt_example/README.rst:100]: Consider having one argument per line, without a consistent style it can be easily confusing for the users to parse this long information.

Comment (dandhlee) [reply samples/api-client/mqtt_example/README.rst:100]: Some arguments are within square brackets `[]` and some are not. They all seem to be optional arguments and I'm confused if they should be the way they are.

File: samples/api-client/mqtt_example/README.rst [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+      --algorithm {RS256,ES256}
+                            Which encryption algorithm to use to generate the JWT.
+      --ca_certs CA_CERTS   CA root from https://pki.google.com/roots.pem
+      --cloud_region CLOUD_REGION
+                            GCP cloud region
+      --data DATA           The telemetry data sent on behalf of a device
+      --device_id DEVICE_ID
+                            Cloud IoT Core device id
+      --gateway_id GATEWAY_ID
+                            Gateway identifier.
```
Comment (dandhlee) [inline samples/api-client/mqtt_example/README.rst:123]: Seeing this block on the [README preview](https://github.com/googleapis/python-iot/tree/mqtt-example-nox/samples/api-client/mqtt_example), it's hard to parse through the information with so much text into this block. I'm not sure what's the best way to present this in a readable way, perhaps adding a line break in between could help, or maybe in chunks of 4 and then a line break.

File: samples/api-client/mqtt_example/cloudiot_mqtt_example.py [hunk 9]
```diff
@@ -111,33 +117,44 @@ def on_disconnect(unused_client, unused_userdata, rc):
...
+    registry_id,
+    device_id,
+    private_key_file,
+    algorithm,
+    ca_certs,
+    mqtt_bridge_hostname,
+    mqtt_bridge_port,
+):
     """Create our MQTT client. The client_id is a unique string that identifies
     this device. For Google Cloud IoT Core, it must be in the format below."""
```
Comment (dandhlee) [inline samples/api-client/mqtt_example/cloudiot_mqtt_example.py:145]: nit: if we're following the Google Docstring style, it should be formatted more like

```python
"""Create our MQTT client.

The client_id is a unique string that identifies this device. For 
Google Cloud IoT Core, it must be in the format below.
"""
```
see https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings.

File: samples/api-client/mqtt_example/noxfile.py [hunk 10]
```diff
@@ -0,0 +1,266 @@
+# Copyright 2019 Google LLC
```
Comment (dandhlee) [inline samples/api-client/mqtt_example/noxfile.py:1]:
```suggestion
# Copyright 2021 Google LLC
```

File: samples/api-client/mqtt_example/noxfile.py [hunk 11]
```diff
@@ -0,0 +1,266 @@
...
+    session.install(BLACK_VERSION)
+    python_files = [path for path in os.listdir(".") if path.endswith(".py")]
+
+    session.run("black", *python_files)
+
+
+#
+# Sample Tests
+#
+
```
Comment (dandhlee) [inline samples/api-client/mqtt_example/noxfile.py:175]: It seems like there are no tests here. If you don't want to work on a test right now I'm fine with this PR as-is, but could there be an issue followed up to make sure there is a test coverage for this sample soon?

**Review by dandhlee — CHANGES_REQUESTED** [review]
Please see my comments below!

As well, `docs` prefix triggers a patch version bump, do we want that happening from this PR?

File: samples/api-client/mqtt_example/README.rst (hunk 1 above)
Comment (gcseh) [reply samples/api-client/mqtt_example/README.rst]: Actually when I [fix](https://github.com/googleapis/python-iot/pull/194/commits/155e70f19b5b7689453be794187111dc0d311ebd) the repo path in the readme template, OwlBot automatically [changes it back](https://github.com/googleapis/python-iot/pull/194/commits/8fe6fbe11974e18aea3c9de3e0baddb6285c9de5) to python-docs-samples. I could not find where the value actually comes from.

File: samples/api-client/mqtt_example/README.rst (hunk 2 above)
Comment (gcseh) [reply samples/api-client/mqtt_example/README.rst]: done

File: samples/api-client/mqtt_example/README.rst (hunk 3 above)
Comment (gcseh) [reply samples/api-client/mqtt_example/README.rst]: done

Comment (gcseh) [reply samples/api-client/mqtt_example/README.rst]: done

File: samples/api-client/mqtt_example/README.rst (hunk 4 above)
Comment (gcseh) [reply samples/api-client/mqtt_example/README.rst]: same as https://github.com/googleapis/python-iot/pull/194#discussion_r715474402

File: samples/api-client/mqtt_example/README.rst (hunk 7 above)
Comment (gcseh) [reply samples/api-client/mqtt_example/README.rst:100]: This whole thing comes from the readme generation, I cannot change it.

File: samples/api-client/mqtt_example/README.rst (hunk 8 above)
Comment (gcseh) [reply samples/api-client/mqtt_example/README.rst:123]: This whole thing comes from the readme generation, I cannot change it.

File: samples/api-client/mqtt_example/cloudiot_mqtt_example.py (hunk 9 above)
Comment (gcseh) [reply samples/api-client/mqtt_example/cloudiot_mqtt_example.py]: done

File: samples/api-client/mqtt_example/noxfile.py (hunk 11 above)
Comment (gcseh) [reply samples/api-client/mqtt_example/noxfile.py:175]: Yes, that will be done longer term. Unfortunately many of these things are leftovers from the migration from python-docs-samples to python-iot repo.

**Comment by dandhlee** [comment]
Discussed offline, closing the loop here; see [example](https://github.com/googleapis/python-iot/commit/59dee60b06536fb01cd4d1df3c0e998cd7dd0108) on adding temporary OwlBot changes to make it happy but also continue receiving template updates when needed.

File: samples/api-client/mqtt_example/noxfile.py (hunk 10 above)
Comment (dandhlee) [reply samples/api-client/mqtt_example/noxfile.py:1]: Please don't forget the year update here!

File: samples/api-client/mqtt_example/noxfile.py (hunk 11 above)
Comment (dandhlee) [reply samples/api-client/mqtt_example/noxfile.py:175]: SGTM. Could you file an internal bug / GitHub issue for it?

**Review by dandhlee — APPROVED** [review]
LGTM, please see two actions items below.

File: samples/api-client/mqtt_example/noxfile.py (hunk 10 above)
Comment (gcseh) [reply samples/api-client/mqtt_example/noxfile.py]: done.

File: samples/api-client/mqtt_example/noxfile.py (hunk 11 above)
Comment (gcseh) [reply samples/api-client/mqtt_example/noxfile.py:175]: done
